### PR TITLE
My Go code for the challenge.

### PR DESCRIPTION
**Pull request by birenderjit**
My solution in Go. Tried to do idiomatic go.

### CODE REVIEW

**Review by ericwburden — COMMENTED** [review]
I'm assuming the three separate files are required. Would it be cleaner to include them all in a `birenderjit.go` folder, maybe?

**Comment by birenderjit** [comment]
I have thought of that option and I think that will make it cleaner. I will also try another option and see if that works, basically name the file as say birenderjit_test.go and include tests and all the code in the same file and see if i can execute the code as tests. 
I believe the idiomatic go way is to have tests separate within the module/package, so that will result in 3 files, a *.go, *_tests.go and the go.mod file.

**Comment by birenderjit** [comment]
Updated to reflect new code structure

**Review by ericwburden — APPROVED** [review]

**Review by xanderyzwich — APPROVED** [review]

File: challenges/2021-05-04-a-box-in-a-box/solutions/birenderjit.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+//Rectangle represents the rectangle type
+type Rectangle struct {
+    leftEdge int
+    rightEdge int
+    topEdge int
+    bottomEdge int
+}
+
+//CreateRectangle is a factory function that creates a new rectangle type
+func CreateRectangle(rd rectangleData) Rectangle {
```
Comment (drkennetz) [inline challenges/2021-05-04-a-box-in-a-box/solutions/birenderjit.go]: subtle difference here in how we construct our rectangles, but I think it looks A okay!

**Review by drkennetz — APPROVED** [review]

**Pull request merged**